### ISSUE TITLE
Automate Release Process

### ISSUE THREAD

**Issue by shinenelson** [ForkAwesome/Fork-Awesome#336]
#335 proves ( again ) that humane error is inevitable. We need to automate our release process. There are [many systems](https://github.com/topics/release) that assist in releasing software on GitHub. The most popular is https://github.com/semantic-release, https://github.com/conventional-changelog and https://github.com/release-it.

Brief introductions : 
- https://github.com/semantic-release automates the whole package release workflow including: determining the next version number, generating the release notes and publishing the package.
This removes the immediate connection between human emotions and version numbers, strictly following the [Semantic Versioning](https://semver.org) specification.
- https://github.com/conventional-changelog generates a **CHANGELOG** from `git` metadata
- https://github.com/release-it is a generic CLI tool to automate versioning and package publishing related tasks

We need to figure out how to automate our release process for future releases.

**Comment by shinenelson** [comment]
I am in favor of https://github.com/semantic-release since we already follow the [Semantic Versioning](https://semver.org) specification.

https://github.com/conventional-changelog only generates the CHANGELOG from commits. I am of the opinion that we cannot enforce commit message styles on contributors. I am, personally, not a fan of the [Angular convention for commit messages](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#-commit-message-guidelines). I am more of a [imperative mood](https://chris.beams.io/posts/git-commit/#imperative) person.

https://github.com/release-it, being a generic command-line-based tool, is a one-for-many tool. It has a plugin-based workflow and possibly would require more integration steps to work for us. This would be my second preference if https://github.com/semantic-release doesn't simply work out of the box and would still require additional _integration_ steps.

**Comment by FWDekker** [comment]
Wow, I'm seeing a lot of activity on the repo! Makes me happy :-)

I'm not sure if this is the best place to put this, but I would also like to suggest creating a release on GitHub rather than just a tag in the future, regardless of whether it has any attachments like packages or whatever. My primary reason for suggesting this is that I watch this repo for new releases so I get a notification, but there is no option to watch for _tags_.

**Comment by shinenelson** [comment]
> I would also like to suggest creating a release on GitHub rather than just a tag in the future, regardless of whether it has any attachments like packages or whatever. My primary reason for suggesting this is that I watch this repo for new releases so I get a notification, but there is no option to watch for _tags_.

Yes, that is the future that we would like to get to too. And that is what we would like to achieve by adopting one of the above mentioned release _frameworks_. If you have other suggestions ( to automate the release process ), please suggest them too.

**Comment by DougInAMug** [comment]
I'm going to look into @semantic-release, if you say that's your favourite option @shinenelson

**Comment by shinenelson** [comment]
I have not used this myself ( hence I do not really how it works ), but I came across [Automatically generated release notes](https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes#configuring-automatically-generated-release-notes) page on the GitHub docs website.

I am note sure whether this helps in making the release itself; but I do think that eases our need to have to figure out _how_ to generate release notes, at least.
